addon/data/viewerworker.js

```diff
@@ -27,6 +27,51 @@ document.addEventListener("error", function (event) {
   self.port.emit("alertError", event.detail);
 }, false);
 
+document.addEventListener("request-account", function (event) {
+  self.port.emit("requestAccount", event.detail);
+});
+
+document.addEventListener("request-profile", function (event) {
+  self.port.emit("requestProfile", event.detail);
+}, false);
+
+document.addEventListener("request-profile-update", function (event) {
+  self.port.emit("requestProfileUpdate", event.detail);
+}, false);
+
+self.port.on("account", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-account", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profile", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-profile", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profileUpdate", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-profile-update", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profileRefresh", function () {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("profile-refresh", true, true, null);
+  document.dispatchEvent(event);
+});
+
```

Comment:
Agreed on simplifying.

* We can ax `got-profile` and `got-profile-update`, since they're taken care of by `profile-refresh` to set the state. If you're already signed in, but the page didn't pick up on that, we can just emit `profile-refresh`.
* Also, `got-account` just tells the page, "Hey, I'm starting the OAuth flow." I thought this might surface errors in the OAuth flow, too, but it doesn't, and they're not user-actionable. So we can ax that.